pennylane/templates/subroutines/multiplier.py

```diff
@@ -269,25 +270,42 @@ def _multiplier_decomposition_resources(
     else:
         num_wires_aux = num_x_wires
 
-    resources = {
-        resource_rep(QFT, num_wires=num_wires_aux): 2,
-        resource_rep(
-            ControlledSequence,
-            base_class=PhaseAdder,
-            base_params={"num_x_wires": num_wires_aux, "mod": mod},
-            num_control_wires=num_x_wires,
-        ): 1,
-        adjoint_resource_rep(
-            ControlledSequence,
-            {
-                "base_class": PhaseAdder,
-                "base_params": {"num_x_wires": num_wires_aux, "mod": mod},
-                "num_control_wires": num_x_wires,
-            },
-        ): 1,
-        adjoint_resource_rep(QFT, {"num_wires": num_wires_aux}): 2,
-        SWAP: num_x_wires,
+    cs_base_params = {
+        "base_class": PhaseAdder,
+        "base_params": {"num_x_wires": num_wires_aux, "mod": mod},
+        "num_control_wires": num_x_wires,
+    }
+
+    params = {
+        "compute_op_params": {
+            "compute_op": resource_rep(QFT, num_wires=num_wires_aux),
+            "target_op": resource_rep(ControlledSequence, **cs_base_params),
+            "uncompute_op": resource_rep(
```

Comment:
Maybe I got lost here, but can't we use `adjoint_resource_rep` directly for this one? 🤔

Reply:
No, you are right, just did that.